plugins/kubernetes/app/decorators/deploy_group_decorator.rb

```diff
@@ -1,12 +1,14 @@
 # frozen_string_literal: true
 DeployGroup.class_eval do
+  has_soft_deletion default_scope: true
   has_one(
     :cluster_deploy_group,
     class_name: 'Kubernetes::ClusterDeployGroup',
     foreign_key: :deploy_group_id,
     inverse_of: :deploy_group
   )
   has_one :kubernetes_cluster, class_name: 'Kubernetes::Cluster', through: :cluster_deploy_group, source: :cluster
+  has_many :kubernetes_deploy_group_roles, class_name: 'Kubernetes::DeployGroupRole'
```

Comment:
use `dependent: :destroy` to make them auto-cleanup

Reply:
Do we need this? this is not triggering anywhere. You mean that we may run destroy manually?

Reply:
It's needed to make cascading deletes work https://github.com/grosser/soft_deletion/blob/master/lib/soft_deletion/core.rb#L17 also it reads nicer/looks safer